src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_star_simplex_iterators.h

```diff
@@ -0,0 +1,285 @@
+    }
+    return (&(sh_->second) == &(other.sh_->second));
+  }
+
+  Simplex_handle const& dereference() const { return sh_; }
+
+  /* Go to the next valid Simplex_handle for a coface, using a breadth first
+   * search approach.
+   *
+   * Invariant:
+   *
+   * sh_ is a coface,
+   * sib_ the Siblings containing sh_,
+   * it_ the root of a coface subtree, such that dim_root_ <= exact_dim_cofaces_.
+   * bfs_queue contains Siblings inside the coface subtree.
+   *
+   * Additionally,
+   *
+   *  - computing all cofaces: sh_ points to a coface of any dimension. bfs_queue
+   * contains a collection of Siblings* that must be considered (as well as there
+   * children). These are all sets of children of simplices in
+   * [ sib_->members().begin(), sh_ ]
+   */
+  void increment_all_cofaces() {
+    ++sh_;  // next sibling
+    // if no more sibling or root of coface tree, go down or to next subtree
+    if (is_root_ || sh_ == sib_->members().end()) {
+      is_root_ = false;
+      if (!bfs_queue_.empty()) {
+        sib_ = bfs_queue_.back();
+        bfs_queue_.pop_back();
+        sh_ = sib_->members().begin();  // don't track dimensions
+        if (st_->has_children(sh_)) {
+          bfs_queue_.insert(bfs_queue_.begin(), st_->children(sh_));
+        }
+      } else {  // bfs_queue == empty, go to root of next coface subtree
+        if (++it_ == end_) {
+          st_ = nullptr;
+          return;
+        }  // no more subtree => end()
+        is_root_ = true;
+        sh_ = *it_;                      // sh_ is the root
+        sib_ = st_->self_siblings(sh_);  // Siblings containing sh_
+        if (st_->has_children(sh_)) {
+          bfs_queue_.insert(bfs_queue_.begin(), st_->children(sh_));
+        }
+        return;  // next root of coface
+      }
+    } else {  // sh_ is valid, simply add its children to the queue
+      if (st_->has_children(sh_)) {
+        bfs_queue_.insert(bfs_queue_.begin(), st_->children(sh_));
+      }
+    }
+  }
+
+  void increment() { increment_all_cofaces(); }
+
+  // Let s be the simplex in a complex C whose star is
+  // iterated through. Let max_v denote the maximal label of vertices in s.
+  SimplexTree* st_;  // Simplex tree for complex C
+  // The cofaces of s form a subforest of the simplex tree. The roots of trees in this
+  // forest have label max_v.
+  //[it_,end_) == range of Simplex_handles of the roots of the cofaces trees (any dim)
+  Simplex_tree_optimized_cofaces_rooted_subtrees_simplex_iterator<SimplexTree> it_;
+  Simplex_tree_optimized_cofaces_rooted_subtrees_simplex_iterator<SimplexTree> end_;
+  // curr Simplex_handle, returned by operator*, pointing to a coface of s
+  Simplex_handle sh_;
+  // set of siblings containing sh_ in the Simplex_tree
+  Siblings* sib_;  //
+  // use a bfs search to avoid calling sib_->members().find(.)
+  // A static_vector is quite faster than a queue
+  // 100 seems a conservative bound on the number of children a Sibling can coutain for now.
```

Comment:
What? The root Siblings can have as many children as there are vertices (minus 1). And even if that one somehow cannot appear in coface computation, the Siblings of the first vertex can have n-2 (you can just add all possible triangles for instance).
If the size of this container is indeed related to Siblings (that would make sense for a BFS), we cannot use a static_vector. If instead it is bounded by the depth of the Simplex_tree (which would probably be the case for a DFS?), we could use 40 like we do elsewhere.

I don't know if there was a good reason for Clément to choose a BFS or if he picked this order arbitrarily.

Reply:
Hum... You are right. From the test I performed, the size was not going up to 20 (with my benchmark), this is why I put 100. I don't know what was Clément intention, but let's go with a std::vector as timings are still good and it will be dynamic.
I don't think it will grow that much as the main code that is used is:
```cpp
      if (!children_stack_.empty()) {
        sib_ = children_stack_.back();
        children_stack_.pop_back();
        sh_ = sib_->members().begin();  // don't track dimensions
        if (st_->has_children(sh_)) {
          children_stack_.push_back(st_->children(sh_));
        }
      }
```
I am using a std::vector from [ca218a9](https://github.com/GUDHI/gudhi-devel/pull/877/commits/ca218a989b250885e8cd8056f425b566b0df9a64)

Reply:
I would say the "main" code is `++sh_` on the first line (iterate on all the nodes of this Siblings) and then the last `else` branch that adds the corresponding children. So it adds all the children to the stack before it starts looking at them (normal for BFS, although it could be done more implicitly) and the structure can contain roughly n_vertices * dimension objects. Of course, this structure will only grow large in cases where one vertex has many neighbors, which is a bad case for this optimization. But at least we want to avoid crashing in that case, even if it will be slow. So yes, vector (or `stack<vector>`).